community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/CostBasedPipeBuilderFactory.scala

```diff
@@ -38,7 +38,9 @@ object CostBasedPipeBuilderFactory {
             tokenResolver: SimpleTokenResolver = new SimpleTokenResolver(),
             maybeExecutionPlanBuilder: Option[PipeExecutionPlanBuilder] = None,
             plannerName: CostBasedPlannerName = PlannerName.default,
-            runtimeName: RuntimeName = RuntimeName.default
+            runtimeName: RuntimeName = RuntimeName.default,
+            useErrorsOverWarnings: Boolean = false,
+            showWarnings: Boolean = false
```

Comment:
I don't really like adding boolean parameters, especially not after each other. Couldn't we send through a function `handleFallback` which will take some enum parameter `severity`, with modes `error`, `warning` and `silence`, and produces an `Option` instead? It would also take the function that produces the fallback plan in case `error` wasn't used.